updater_test.go

```diff
@@ -0,0 +1,191 @@
+	logger := &logrTesting.TestLogger{T: t}
+
+	ns1 := &corev1.Namespace{
+		TypeMeta: v1.TypeMeta{Kind: "Namespace", APIVersion: "v1"},
+		ObjectMeta: v1.ObjectMeta{
+			Name: "ns1",
+		},
+	}
+
+	pod1 := &corev1.Pod{
+		TypeMeta: v1.TypeMeta{Kind: "Pod", APIVersion: "v1"},
+		ObjectMeta: v1.ObjectMeta{
+			Name:      "pod1",
+			Namespace: "ns1",
+			Annotations: map[string]string{
+				TelegrafClass:       "test",
+				TelegrafMetricsPath: "/metrics",
+				TelegrafMetricsPort: "6060",
+			},
+		},
+	}
+
+	pod2 := &corev1.Pod{
+		TypeMeta: v1.TypeMeta{Kind: "Pod", APIVersion: "v1"},
+		ObjectMeta: v1.ObjectMeta{
+			Name:      "pod2",
+			Namespace: "ns1",
+			Annotations: map[string]string{
+				TelegrafClass:       "app",
+				TelegrafMetricsPath: "/metrics",
+				TelegrafMetricsPort: "6060",
+			},
+		},
+	}
+
+	secret1 := &corev1.Secret{
+		TypeMeta: v1.TypeMeta{Kind: "Pod", APIVersion: "v1"},
+		ObjectMeta: v1.ObjectMeta{
+			Name:      "telegraf-config-pod1",
+			Namespace: "ns1",
+			Labels: map[string]string{
+				TelegrafSecretLabelClassName: "test",
+				TelegrafSecretLabelPod:       pod1.GetObjectMeta().GetName(),
+			},
+		},
+		Data: map[string][]byte{
+			TelegrafSecretDataKey: []byte("ns1/pod1/test"),
+		},
+	}
+
+	secret2 := &corev1.Secret{
+		TypeMeta: v1.TypeMeta{Kind: "Pod", APIVersion: "v1"},
+		ObjectMeta: v1.ObjectMeta{
+			Name:      "telegraf-config-pod2",
+			Namespace: "ns1",
+			Labels: map[string]string{
+				TelegrafSecretLabelClassName: "app",
+				TelegrafSecretLabelPod:       pod2.GetObjectMeta().GetName(),
+			},
+		},
+		Data: map[string][]byte{
+			TelegrafSecretDataKey: []byte("ns1/pod2/app"),
+		},
+	}
+
+	secret3 := &corev1.Secret{
+		TypeMeta: v1.TypeMeta{Kind: "Pod", APIVersion: "v1"},
+		ObjectMeta: v1.ObjectMeta{
+			Name:      "unrelated1",
+			Namespace: "ns1",
+			Labels:    map[string]string{},
+		},
+		Data: map[string][]byte{
+			"key": []byte("value"),
+		},
+	}
+
+	return &secretsUpdaterTest{
+		logger:  logger,
+		ns1:     ns1,
+		pod1:    pod1,
+		pod2:    pod2,
+		secret1: secret1,
+		secret2: secret2,
+		secret3: secret3,
+	}
+}
+
+// createObjects creates Kubernetes fake clientset as well as other objects that depend on it.
+func (t *secretsUpdaterTest) createObjects() {
+	t.fakeClient = fake.NewSimpleClientset(
+		t.ns1,
+		t.pod1, t.pod2,
+		t.secret1, t.secret2, t.secret3,
+	)
+
+	t.mockSidecar = &mockSidecarHandler{}
+
+	t.updater = &secretsUpdater{
+		logger:    t.logger,
+		clientset: t.fakeClient,
+		sidecar:   t.mockSidecar,
+	}
+
+}
+
+func Test_AssembleConfForSecretsWithLabels(t *testing.T) {
+	test := newSecretsUpdaterTest(t)
+
+	test.createObjects()
+	test.updater.onChange()
+	// validate that assembleConf() was called for secret1 and secret2, but not secret3
+	if want, got := "ns1/pod1/test;ns1/pod2/app", strings.Join(test.mockSidecar.get(), ";"); want != got {
+		t.Errorf("wrong configurations assembled; want=%v; got=%v", want, got)
```

Comment:
I prefer rendering quoted strings; it's easier to spot trailing spaces etc

```suggestion
		t.Errorf("wrong configurations assembled; want=%q; got=%q", want, got)
```